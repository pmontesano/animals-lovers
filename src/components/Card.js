import React from 'react';

const Card = ({ photo, alt, title, action, handleClick, id }) => {
  const handleButtonClick = (id) => (e) => {
    handleClick(id);
  };

  return (
    <div className="card">
      <div className="card-photo">
        <img className="mx-auto" src={photo} alt={alt} />
      </div>

      <div className="px-6 py-4">
        <div className={`${title ? 'font-bold text-xl mb-2' : 'skeleton-box'}`}>
          {title}
        </div>
      </div>
      <div className="px-6 py-4">
        {action && (
          <button
            className="btn btn-blue btn-blue:hover"
            onClick={handleButtonClick(id)}
          >
            {action}
          </button>
        )}
      </div>
    </div>
  );
};

export default Card;
