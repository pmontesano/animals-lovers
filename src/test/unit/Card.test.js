import React from 'react';
import { create } from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Card from '../../components/Card';

configure({ adapter: new Adapter() });

const handleButtonClick = jest.fn();

const props = {
  photo: 'somephoto.svg',
  alt: 'photo alt',
  title: 'title',
  action: true,
  id: '1234',
  handleClick: handleButtonClick,
};

describe('Card', () => {
  it('should render correctly', () => {
    const wrapper = create(<Card {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should be call handleButtonClick', () => {
    const wrapper = shallow(<Card {...props} />);
    wrapper.find('button').at(0).simulate('click');
    expect(handleButtonClick.mock.calls.length).toEqual(1);
  });
});
