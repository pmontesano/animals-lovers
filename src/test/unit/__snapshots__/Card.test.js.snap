// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-photo"
  >
    <img
      alt="photo alt"
      className="mx-auto"
      src="somephoto.svg"
    />
  </div>
  <div
    className="px-6 py-4"
  >
    <div
      className="font-bold text-xl mb-2"
    >
      title
    </div>
  </div>
  <div
    className="px-6 py-4"
  >
    <button
      className="btn btn-blue btn-blue:hover"
      onClick={[Function]}
    />
  </div>
</div>
`;
